docs/getting-started.md

```diff
@@ -0,0 +1,325 @@
+You can restrict which users are allowed to login with `Authenticator.whitelist`:
+
+```python
+c.Authenticator.whitelist = {'mal', 'zoe', 'inara', 'kaylee'}
+```
+
+After starting the server, you can add and remove users in the whitelist via the `admin` panel,
+which brings us to...
+
+```python
+c.JupyterHub.admin_users = {'mal', 'zoe'}
+```
+
+Any users in the admin list are automatically added to the whitelist, if they are not already present.
+
+Admin users have the ability to take actions on users' behalf,
+such as stopping and restarting their servers, and adding and removing new users.
+If `JupyterHub.admin_access` is True (not default),
+then admin users have permission to log in *as other users* on their respective machines,
+for debugging. **You should make sure your users know if admin_access is enabled.**
+
+### adding and removing users
+
+The default PAMAuthenticator is one case of a special kind of authenticator,
+called a LocalAuthenticator,
+indicating that it manages users on the local system.
+When you add a user to the Hub, a LocalAuthenticator checks if that user already exists.
+Normally, there will be an error telling you that the user doesn't exist.
+If you set the config value
+
+```python
+c.LocalAuthenticator.create_system_users = True
+```
+
+however, adding a user to the Hub that doesn't already exist on the system will result
+in the Hub creating that user via the system `useradd` mechanism.
+This option is typically used on hosted deployments of JupyterHub,
+to avoid the need to manually create all your users before launching the service.
+It is not recommended when running JupyterHub on 'real' machines with regular users.
+
+
+## Configuring single-user servers
+
+Since the single-user server is an instance of `ipython notebook`,
+an entire separate multi-process application,
+there is a lot you can configure,
+and a lot of ways to express that configuration.
+
+At the JupyterHub level, you can set some values on the Spawner.
+The simplest of these is `Spawner.notebook_dir`,
+which lets you set the root directory for a user's server.
+`~` is expanded to the user's home directory.
+
+```python
+c.Spawner.notebook_dir = '~/notebooks'
+```
+
+You can also specify extra command-line arguments to the notebook server with
+
+```python
+c.Spawner.args = ['--debug', '--profile=PHYS131']
+```
+
+Since the single-user server extends the notebook server application,
+it still loads configuration from the `ipython_notebook_config.py` config file.
+Each user may have one of these files in `$HOME/.ipython/profile_default/`.
+IPython also supports loading system-wide config files from `/etc/ipython/`,
+which is the place to put configuration that you want to affect all of your users.
+
+
+- setting working directory
+- setting default page
+- /etc/ipython
+- custom Spawner
+
+## external services
+
+JupyterHub has a REST API that can be used
```

Comment:
ha, yes. I might remove the external services bit from this doc. I was originally thinking of documenting using `jupyterhub token` and setting up the culling script, but I might save that for a more dedicated doc.